### PR TITLE
New earn page

### PR DESCRIPTION
**Pull request by solovev**

### CODE REVIEW

File: app/(landing)/views/offer/token.tsx [hunk 1]
```diff
@@ -5,134 +5,189 @@ import Image from 'next/image'
...
-            <div className={styles.content}>
-                <Logo {...{ token, development }} />
+export function Token({ token, children, development, balance, href, buttonLabel }: PropsWithChildren<TokenProps>) {
+  return (
+    <div className={clsx(styles.token, styles[token])}>
+      <TokenContext.Provider value={{ token, development }}>
+        <div className={styles.content}>
+          <Logo {...{ token, development }} />
 
-                {children}
```
Comment (LeoVS09) [inline app/(landing)/views/offer/token.tsx:32]: Why does PR change indentation? I think we have linter and editorconfig that define exact indendation

File: app/(landing)/views/offer/token.tsx [hunk 2]
```diff
@@ -5,134 +5,189 @@ import Image from 'next/image'
...
+          value={100}
+          classNames={{
+            indicator: styles.insuranceInducator,
+          }}
+        />
+      </div>
+
+      {href ? <InternalLink href={href}>{<CTAButton />}</InternalLink> : <CTAButton />}
+
+      {development && <p className={styles.development}>Coming soon</p>}
```
Comment (LeoVS09) [inline app/(landing)/views/offer/token.tsx:164]: Better move this label to other place, it looks like a button, but it not a button and it not a clickable. If it label, it must looks like a label. Check design for example
image

File: app/earn/components/vault-card/vault-user-balance.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  const { value: price, decimals: priceDecimals } = vault.asset.price
+  const threshold = BigInt(1e6) * 10n ** BigInt(decimals)
+
+  return (
+    (walletStatus === WalletStatus.CONNECTING || isLoading) ? <Loader /> : <Value />
+  )
+
+  function Value() {
+    if (walletStatus === WalletStatus.NOT_CONNECTED) {
+      return '-'
```
Comment (LeoVS09) [inline app/earn/components/vault-card/vault-user-balance.tsx:28]: In such case better not display anything at all

**Review by LeoVS09 — COMMENTED** [review]
Looks very good, except few small issues

File: app/(landing)/views/offer/token.tsx (hunk 1 above)
Comment (solovev) [reply app/(landing)/views/offer/token.tsx:32]: idk, i use eslint formatter

File: app/(landing)/views/offer/token.tsx (hunk 2 above)
Comment (solovev) [reply app/(landing)/views/offer/token.tsx:166]: Stop adding new elements to this card please, i disabled the button

File: app/earn/components/vault-card/vault-user-balance.tsx (hunk 3 above)
Comment (solovev) [reply app/earn/components/vault-card/vault-user-balance.tsx]: ok

File: app/(landing)/views/offer/token.tsx (hunk 1 above)
Comment (LeoVS09) [reply app/(landing)/views/offer/token.tsx:32]: Are you have another config in IDE?

**Review by LeoVS09 — APPROVED** [review]
You can merge, please remove line between balance and insurance if you will have time

File: app/(landing)/views/offer/token.tsx (hunk 1 above)
Comment (solovev) [reply app/(landing)/views/offer/token.tsx:32]: no, i have no customs there

**Pull request merged**